baseplate/integration/mixins/lazy_attributes.py

```diff
@@ -0,0 +1,22 @@
+class LazyAttributesMixin(object):
+    """A mixin that allows instances to add a lazily evaluated attribute
+       through `add_lazy_property`.
+    """
+
+    # Called when an attribute is missing
+    def __getattr__(self, name):
+        if name not in self._lazy_attributes:
```

Comment:
since we're trying to shave off microseconds here, i wonder if only touching the dictionary once would help?

```python
try:
    lazy_fn = self._lazy_attributes[name]
except KeyError:
    return super(LazyAttributesMixin, self).__getattr__(name)

prop_value = lazy_fn(self)
setattr(self, name, prop_value)
return prop_value
```